src/Microsoft.SqlTools.ServiceLayer/Rename/RenameService.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Rename.Requests;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename
+{
+    /// <summary>
+    /// Main class for Rename Service functionality
+    /// </summary>
+    public class RenameService : IDisposable
+    {
+        private bool disposed = false;
+
+        private static Lazy<RenameService> renameServiceInstance = new Lazy<RenameService>(() => new RenameService());
+        public static RenameService Instance => renameServiceInstance.Value;
+
+        public static ConnectionService connectionService;
+
+        private IProtocolEndpoint serviceHost;
+
+        public RenameService() { }
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(ProcessRenameEditRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            // The request handling will take some time to return, we need to use a separate task to run the request handler so that it won't block the main thread.
+            // For any specific table designer instance, ADS UI can make sure there are at most one request being processed at any given time, so we don't have to worry about race conditions.
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(ProcessRenameEditRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in ProcessRenameEditRequest()");
+               bool operationExecutedSuccessFull = false;
```

Comment:
Remove this - we only ever send the result if it succeeded so you can just hardcode it to `SendResult(true)` below